build/gen_stub.php

```diff
@@ -1200,20 +1201,16 @@ private function setRefcount(?string $refcount): void
         $isScalarType = $type !== null && $type->isScalar();
 
         if ($refcount === null) {
-            $this->refcount = $isScalarType ? self::REFCOUNT_0 : self::REFCOUNT_N;
+            $this->refcount = $isScalarType ? self::REFCOUNT_SCALAR : self::REFCOUNT_N;
             return;
         }
 
-        if (!in_array($refcount, ReturnInfo::REFCOUNTS, true)) {
-            throw new Exception("@refcount must have one of the following values: \"0\", \"1\", \"N\", $refcount given");
+        if ($isScalarType) {
+            throw new Exception("@refcount is not permitted on functions returning scalar values");
```

Comment:
maybe the error message could also indicate that adding `@refcount 0` is superfluous?

Reply:
Sure, done